### ISSUE TITLE
Add `StringViewArray` implementation and layout and basic construction + tests

### ISSUE THREAD

**Issue by alamb** [apache/arrow-rs#5469]
This is part of the larger project to implement `StringViewArray` -- see https://github.com/apache/arrow-rs/issues/5374

After https://github.com/apache/arrow-rs/issues/5468 we will next need to implement `StringViewArray` 

For inspiration I think you can look at https://github.com/apache/arrow-rs/pull/4585 

(specifically `arrow-array/src/array/byte_view_array.rs` https://github.com/apache/arrow-rs/pull/4585/files#diff-160ecd8082d5d28081f01cdb08a898cb8f49b17149c7118bf96746ddaae24b4f)

Basic tasks:
1. Create`Utf8ViewArray` and implement `Array` for it. This should follow the outline from https://github.com/apache/arrow-rs/pull/4585 and implement a similar  API to  `StringArray` https://docs.rs/arrow/latest/arrow/array/type.StringArray.html
2. Add documentaton for new array, with documentation 
3. Examples of constructing the array in docs
4. Tests for basic creation APIs (like `new()` and `new_unchecked`, `value()`, `is_null()` etc). 

Potentially: implement pretty printing for `Utf8ViewArray` via [`ArrayFormatter`](https://docs.rs/arrow/latest/arrow/util/display/struct.ArrayFormatter.html), though this could be done as a separate ticket / PR.

**Comment by ariesdevil** [comment]
Please assign me, thx :)

**Comment by alamb** [comment]
I plan to merge https://github.com/apache/arrow-rs/pull/5470 later today -- I don't expect substantial changes, so if you would like to start on this issue, you could probably make a branch from https://github.com/apache/arrow-rs/pull/5470 and rebase once we merge that

**Comment by alamb** [comment]
Noce that https://github.com/apache/arrow-rs/pull/5470 is merged I think this ticket is unblocked and ready to go

Note I think this ticket will be somewhat of a bottleneck. Once it is done I think we can start implementing other features in parallel (like IPC support, cast, filter, etc)

**Comment by tustvold** [comment]
`label_issue.py` automatically added labels {'arrow'} from #5481

**Comment by tustvold** [comment]
`label_issue.py` automatically added labels {'parquet'} from #4585
